x/rewards/mintbankkeeper/keeper.go

```diff
@@ -0,0 +1,78 @@
+package mintbankkeeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authTypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	mintTypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+
+	"github.com/archway-network/archway/pkg"
+	rewardsTypes "github.com/archway-network/archway/x/rewards/types"
+)
+
+var _ mintTypes.BankKeeper = Keeper{}
+
+// RewardsKeeperExpected defines the expected interface for the x/rewards keeper.
+type RewardsKeeperExpected interface {
+	InflationRewardsEnabled(ctx sdk.Context) bool
+	InflationRewardsRatio(ctx sdk.Context) sdk.Dec
+	TrackInflationRewards(ctx sdk.Context, rewards sdk.Coin)
+}
+
+// Keeper is the x/bank keeper decorator that is used by the x/mint module.
+// Decorator is used to split inflation tokens between the rewards collector and the fee collector accounts.
+type Keeper struct {
+	bankKeeper    mintTypes.BankKeeper
+	rewardsKeeper RewardsKeeperExpected
+}
+
+// NewKeeper creates a new Keeper instance.
+func NewKeeper(bk mintTypes.BankKeeper, rk RewardsKeeperExpected) Keeper {
+	return Keeper{
+		bankKeeper:    bk,
+		rewardsKeeper: rk,
+	}
+}
+
+// SendCoinsFromModuleToModule implements the mintTypes.BankKeeper interface.
+func (k Keeper) SendCoinsFromModuleToModule(ctx sdk.Context, senderModule, recipientModule string, amt sdk.Coins) error {
+	// Perform the split only if the recipient is fee collector (which for instance is always the case) and
+	// inflation rewards are enabled.
+	if recipientModule != authTypes.FeeCollectorName || !k.rewardsKeeper.InflationRewardsEnabled(ctx) {
+		return k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, recipientModule, amt)
+	}
+
+	ratio := k.rewardsKeeper.InflationRewardsRatio(ctx)
```

Comment:
We're fetching inflationary rewards twice, we could add here the logic of `IsZero()`.

Reply:
Done